# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../blackbox/gcc43.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- blackbox-0.70.1/lib/Image.cc.vanilla	2008-04-09 09:50:10.000000000 +0000
+++ blackbox-0.70.1/lib/Image.cc	2008-04-09 09:50:24.000000000 +0000
@@ -27,6 +27,8 @@
 #include "Pen.hh"
 #include "Texture.hh"
 
+#include <string.h>
+
 #include <algorithm>
 #include <vector>
 
--- blackbox-0.70.1/src/Screen.cc.vanilla	2008-04-09 09:52:23.000000000 +0000
+++ blackbox-0.70.1/src/Screen.cc	2008-04-09 09:52:36.000000000 +0000
@@ -48,6 +48,8 @@
 #include <ctype.h>
 #include <dirent.h>
 
+#include <string.h>
+
 
 static bool running = true;
 static int anotherWMRunning(Display *, XErrorEvent *) {
--- blackbox-0.70.1/src/main.cc.vanilla	2008-04-09 09:52:56.000000000 +0000
+++ blackbox-0.70.1/src/main.cc	2008-04-09 09:53:02.000000000 +0000
@@ -35,6 +35,7 @@
 #include "../version.h"
 
 #include <stdio.h>
+#include <string.h>
 
 
 static void showHelp(int exitval) {
--- blackbox-0.70.1/src/BlackboxResource.cc.vanilla	2008-04-09 09:51:35.000000000 +0000
+++ blackbox-0.70.1/src/BlackboxResource.cc	2008-04-09 09:51:43.000000000 +0000
@@ -33,6 +33,7 @@
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
 
+#include <string.h>
 
 BlackboxResource::BlackboxResource(const std::string& rc): rc_file(rc) {
   screen_resources = 0;
--- blackbox-0.70.1/src/ScreenResource.cc.vanilla	2008-04-09 09:52:03.000000000 +0000
+++ blackbox-0.70.1/src/ScreenResource.cc	2008-04-09 09:52:19.000000000 +0000
@@ -32,7 +32,7 @@
 #include <Resource.hh>
 
 #include <assert.h>
-
+#include <string.h>
 
 static const int iconify_width  = 9;
 static const int iconify_height = 9;
--- blackbox-0.70.1/util/bsetroot.cc.vanilla	2008-04-09 09:53:21.000000000 +0000
+++ blackbox-0.70.1/util/bsetroot.cc	2008-04-09 09:54:02.000000000 +0000
@@ -30,7 +30,10 @@
 #include <cctype>
 
 #include <X11/Xatom.h>
+
+#include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
 
 // ignore all X errors
--- blackbox-0.70.1/lib/XDG.cc.vanilla	2008-04-09 09:55:46.000000000 +0000
+++ blackbox-0.70.1/lib/XDG.cc	2008-04-09 09:55:56.000000000 +0000
@@ -27,6 +27,8 @@
 
 #include <stdlib.h>
 
+#include <algorithm>
+
 
 // make sure directory names end with a slash
 static std::string terminateDir(const std::string &string)
--- blackbox-0.70.1/lib/Resource.cc.vanilla	2008-04-09 09:55:29.000000000 +0000
+++ blackbox-0.70.1/lib/Resource.cc	2008-04-09 09:55:37.000000000 +0000
@@ -29,6 +29,7 @@
 #include <X11/Xresource.h>
 
 #include <stdio.h>
+#include <string.h>
 
 
 bt::Resource::Resource(void)
